mrp_production_batch/models/mrp_production_batch.py

```diff
@@ -124,22 +121,35 @@ def _compute_product_ids(self):
 
     def action_done(self):
         self.ensure_one()
-        productions = self.production_ids.filtered(
-            lambda r: r.state not in ("cancel", "done")
-        )
+        productions = self.production_ids.filtered(lambda r: r.state != "cancel")
```

Comment:
Process all productions here and check before if all of them are in "ready" state